### PR TITLE
fix: simplify useTokenPrice

### PR DESCRIPTION
**Pull request by UncleBill**

### CODE REVIEW

File: packages/maskbook/src/plugins/Wallet/hooks/useTokenPrice.ts [hunk 1]
```diff
@@ -15,47 +15,50 @@ const task = pollingTask(
         delay: UPDATE_CHAIN_STATE_DELAY,
     },
 )
-
 export function useTokenPrice(
-    platformOrCoinId: string,
+    chainId: ChainId,
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/hooks/useTokenPrice.ts:19]: Nice!

**Review by guanbinrui — APPROVED** [review]

**Comment by septs** [comment]
plz resolve conflict.

**Comment by UncleBill** [comment]
Please DO NOT review on a draft PR, unless the submitter send a review request.

A draft is a drat, that means it's not ready to review.

File: packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx [hunk 2]
```diff
@@ -240,11 +238,8 @@ const ContractInteraction = memo(() => {
     const tokenDecimals = (isNativeTokenInteraction ? nativeToken?.decimals : token?.decimals) ?? 0
 
     // token estimated value
-    const tokenPrice = useTokenPrice(
-        getCoingeckoPlatformId(chainId) ?? '',
-        !isNativeTokenInteraction ? token?.address : undefined,
-    )
-    const nativeTokenPrice = useNativeTokenPrice(getCoingeckoCoinId(nativeToken?.symbol.toLowerCase() ?? '') ?? '')
+    const tokenPrice = useTokenPrice(chainId, !isNativeTokenInteraction ? token?.address : undefined)
+    const nativeTokenPrice = useNativeTokenPrice(nativeToken?.chainId ?? 0)
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx:242]: `0` is not a valid chain id. 😔

File: packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx [hunk 3]
```diff
@@ -84,7 +84,7 @@ export const GasSetting1559 = memo(() => {
     const history = useHistory()
     const [selected, setOption] = useState<number | null>(null)
     const { value: nativeToken } = useNativeTokenDetailed()
-    const nativeTokenPrice = useNativeTokenPrice(getCoingeckoCoinId(nativeToken?.symbol.toLowerCase() ?? '') ?? '')
+    const nativeTokenPrice = useNativeTokenPrice(nativeToken?.chainId ?? 0)
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx:87]: Ditto.

File: packages/maskbook/src/components/shared/TokenPrice.tsx [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
     amount: number
     currencyType?: CurrencyType
 }
 
-export const TokenPrice = ({ platform, contractAddress, amount, currencyType = CurrencyType.USD }: TokenPriceProps) => {
-    const price = useTokenPrice(platform, contractAddress, currencyType) * amount
+export const TokenPrice = ({ chainId, contractAddress, amount, currencyType = CurrencyType.USD }: TokenPriceProps) => {
+    const price = useTokenPrice(chainId, contractAddress, currencyType) * amount
     return <span>${price.toFixed(2)}</span>
 }
```
Comment (septs) [inline packages/maskbook/src/components/shared/TokenPrice.tsx:14]:
```suggestion
export const TokenPrice = (props: TokenPriceProps) => {
    const price = useTokenPrice(props.chainId, props.contractAddress, props.currencyType ?? CurrencyType.USD) * amount
    return <span>${props.price.toFixed(2)}</span>
}
```

File: packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx (hunk 2 above)
Comment (UncleBill) [reply packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx:242]: Then will fail to get valid platformId/coinId, and price will be `0`

**Review by nuanyang233 — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Review by nuanyang233 — APPROVED** [review]

**Pull request merged**